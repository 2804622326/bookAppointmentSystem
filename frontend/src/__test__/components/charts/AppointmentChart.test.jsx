import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import AppointmentChart from '../../../components/charts/AppointmentChart.jsx';

jest.mock('recharts', () => ({
  ResponsiveContainer: ({ children }) => <div>{children}</div>,
}));

jest.mock('../../../components/charts/CustomPieChart.jsx', () => () => <div data-testid='pie' />);

jest.mock('../../../components/appointment/AppointmentService.js', () => ({
  getAppointmentsSummary: jest.fn(() => Promise.resolve({ data: [{ name: 'a', value: 1 }] })),
}));

jest.mock('../../../components/common/NoDataAvailable.jsx', () => () => <div data-testid='no-data' />);

test('shows appointment chart with data', async () => {
  const { asFragment } = render(<AppointmentChart />);
  await waitFor(() => screen.getByText(/appointments overview/i));
  expect(screen.getByTestId('pie')).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
