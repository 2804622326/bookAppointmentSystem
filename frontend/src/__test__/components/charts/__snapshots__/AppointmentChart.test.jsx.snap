// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows appointment chart with data 1`] = `
<DocumentFragment>
  <section>
    <h5
      class="mb-4 chart-title"
    >
      Appointments Overview
    </h5>
    <div
      data-testid="pie"
    />
  </section>
</DocumentFragment>
`;
